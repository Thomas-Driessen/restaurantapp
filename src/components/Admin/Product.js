import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Grid from '@material-ui/core/Grid';
import CloseIcon from '@material-ui/icons/Close';

const Product = (props) => {
    const [open, setOpen] = React.useState(false);
    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    return(
        <div>
            { props.product ? (
                <div>
                <Card >
                    <CardMedia style={{height: 400}}
                    component="img"
                    height="250"
                    src={`../../images/${props.productType}/${props.product.id}.jpg`}
                    alt={`Image for ${props.product.title} Not Found`}
                    title={props.product.title}
                    />
                    <CardContent>
                    <Typography gutterBottom variant="inherit" component="h2">
                        {props.product.title} <span style={{float: "right", color: "green"}}>{props.product.price}€</span>
                    </Typography>
                    <Typography component="h6">
                        {props.product.ingredients}
                    </Typography>
                    </CardContent>
                    <CardActions>
                    <Grid container alignItems="flex-start" justify="flex-end" direction="row">
                    <div style={{display: 'flex', alignItems: 'right'}}>
                    <Button size="large" color="primary" target="_blank" onClick={handleClickOpen}>
                        <span style={{fontWeight:"bold"}}>Edit</span>
                    </Button>
                    <Button size="large" color="primary" target="_blank">
                        <span style={{fontWeight:"bold"}}>Delete</span>
                    </Button>
                    <Dialog
                        open={open}
                        onClose={handleClose}
                        aria-labelledby="product-title"
                        aria-describedby="product-description"
                    >
                        <DialogTitle id="product-title">{props.product.title}</DialogTitle>
                        <DialogContent>
                            <DialogContentText id="product-description">
                                {props.product.description}
                            </DialogContentText>
                            <DialogContentText id="product-description">
                            <span style={{float: "right", color: "green", fontWeight: "bold"}}>{props.product.price}€</span>
                            </DialogContentText>
                        </DialogContent>
                        <DialogActions>
                            <IconButton aria-label="close" color="primary" onClick={handleClose}>
                                <CloseIcon />
                            </IconButton>
                        </DialogActions>
                    </Dialog>
                    </div>
                    </Grid>
                    </CardActions>
                </Card>
                </div>
            ) : null}
        </div>
    )
}
export default Product