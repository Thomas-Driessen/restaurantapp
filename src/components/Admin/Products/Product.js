import React from "react";
import Card from "@material-ui/core/Card";
import clsx from "clsx";
import { makeStyles } from "@material-ui/core/styles";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import ProductEdit from './ProductEdit';
import ProductDelete from './ProductDelete'

const useStyles = makeStyles(() => ({
  blur: {
    filter: "grayscale(80%)",
    transition: "filter .1s",
    '&:hover': { filter: "grayscale(0%)" }
  }
}));

const Product = (props) => {
  const classes = useStyles();

  function renderIngredients() {
    let ingredients = [];
    props.product.ingredients.map(currentIngredient => (
      ingredients.push(currentIngredient.ingredient.ingredientTitle)
    ));

    return ingredients.toString();
  }

  return (
    <div>
      {props.product ? (
        <div>
          <Card className={clsx({ [classes.blur]: !props.product.onMenu })}>
            <CardMedia
              style={{ height: 400 }}
              component="img"
              height="250"
              src={props.product.image}
              alt={`Image for ${props.product.title} Not Found`}
              title={props.product.title}
            />
            <CardContent>
              <Typography gutterBottom variant="inherit" component="h2">
                {props.product.title}{" "}
                <span style={{ float: "right", color: "green" }}>
                  {props.product.price}â‚¬
                </span>
              </Typography>
              <Typography component="h6">
                {renderIngredients()}
              </Typography>
            </CardContent>
            <CardActions>
              <Grid
                container
                alignItems="flex-start"
                justify="flex-end"
                direction="row"
              >
                <div style={{ display: "flex", alignItems: "right" }}>
                  <ProductEdit
                    product={props.product}
                    foodCategories={props.foodCategories}
                    drinkCategories={props.drinkCategories}
                    productType={props.productType}
                    ingredients={props.ingredients}
                  />
                  <ProductDelete
                    product={props.product}
                  />
                </div>
              </Grid>
            </CardActions>
          </Card>
        </div>
      ) : null}
    </div>
  );
};
export default Product;
