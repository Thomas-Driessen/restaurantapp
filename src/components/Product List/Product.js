import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import FavoriteIcon from '@material-ui/icons/Favorite';
import IconButton from '@material-ui/core/IconButton';
import AddIcon from '@material-ui/icons/Add';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import currentFoodList from '../Current Order/CurrentFoodList';
import currentDrinkList from '../Current Order/CurrentDrinkList';
import Grid from '@material-ui/core/Grid';
import CloseIcon from '@material-ui/icons/Close';

const Product = (props) => {
    const [open, setOpen] = React.useState(false);
    const [foodLike, setFoodLike] = React.useState(localStorage.getItem('foodLikes') ? localStorage.getItem('foodLikes').indexOf(props.product.id) > -1 ? true : false : false)
    const [drinkLike, setDrinkLike] = React.useState(localStorage.getItem('drinkLikes') ? localStorage.getItem('drinkLikes').indexOf(props.product.id) > -1 ? true : false : false)

    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    const addFoodLikes = (e) => {
        e.preventDefault();
        let foodLikeNew = {
            food: props.product,
            likes: props.product.likes
        }
        let likes = localStorage.getItem('foodLikes') ? localStorage.getItem('foodLikes').split(',') : [];
        if (foodLike) {
            let index = likes.indexOf(props.product.id);
            likes.splice(index, 1);
            foodLikeNew.likes--;
            setFoodLike(false);
        }
        else {
            likes.push(props.product.id);
            foodLikeNew.likes++;
            setFoodLike(true);
        }

        fetch(`${process.env.REACT_APP_API_URL}/api/FoodLikes`, {
            method: "PUT",
            mode: "cors",
            headers: {
                Accept: "application/json",
                "Content-Type": "application/json",
            },
            body: JSON.stringify(foodLikeNew),
        }).catch(console.log);

        localStorage.setItem('foodLikes', likes);
    }

    const addDrinkLikes = (e) => {
        e.preventDefault();
        let drinkLikeNew = {
            drink: props.product,
            likes: props.product.likes
        }
        let likes = localStorage.getItem('drinkLikes') ? localStorage.getItem('drinkLikes').split(',') : [];
        if (drinkLike) {
            let index = likes.indexOf(props.product.id);
            likes.splice(index, 1);
            drinkLikeNew.likes--;
            setDrinkLike(false);
        }
        else {
            likes.push(props.product.id);
            drinkLikeNew.likes++;
            setDrinkLike(true);
        }

        fetch(`${process.env.REACT_APP_API_URL}/api/DrinkLikes`, {
            method: "POST",
            mode: "cors",
            headers: {
                Accept: "application/json",
                "Content-Type": "application/json",
            },
            body: JSON.stringify(drinkLikeNew),
        }).catch(console.log);

        localStorage.setItem('drinkLikes', likes);
    }

    function addToOrder(e) {
        e.preventDefault();
        if (props.productType === "Food") {
            currentFoodList.push(props.product);
            sessionStorage.setItem("currentFoodList", JSON.stringify(currentFoodList));
        }
        else {
            currentDrinkList.push(props.product);
            sessionStorage.setItem("currentDrinkList", JSON.stringify(currentDrinkList));
        }
    };

    function renderIngredients() {
        let ingredients = [];
        props.product.ingredients.map(currentIngredient => (
            ingredients.push(currentIngredient.ingredient.ingredientTitle)
        ));

        return ingredients.toString();
    }

    return (
        <div>
            { props.product ? (
                <div>
                    <Card >
                        <CardMedia
                            style={{ height: 400 }}
                            component="img"
                            height="250"
                            src={props.product.image}
                            alt={`Image for ${props.product.title} Not Found`}
                            title={props.product.title}
                        />
                        <CardContent>
                            <Typography gutterBottom variant="inherit" component="h2">
                                {props.product.title} <span style={{ float: "right", color: "green" }}>{props.product.price}€</span>
                            </Typography>
                            <Typography component="h6">
                                {renderIngredients()}
                            </Typography>
                        </CardContent>
                        <CardActions>
                            <Grid
                                container
                                alignItems="flex-start"
                                justify="flex-end"
                                direction="row"
                            >
                                <div style={{ display: 'flex', alignItems: 'right' }}>
                                    <Button size="large" color="primary" target="_blank" onClick={handleClickOpen}>
                                        <span style={{ fontWeight: "bold" }}>View Details</span>
                                    </Button>
                                    <Dialog
                                        open={open}
                                        onClose={handleClose}
                                        aria-labelledby="product-title"
                                        aria-describedby="product-description"
                                    >
                                        <DialogTitle id="product-title">{props.product.title}</DialogTitle>
                                        <DialogContent>
                                            <DialogContentText id="product-description">
                                                {props.product.description}
                                            </DialogContentText>
                                            <DialogContentText id="product-description">
                                                <span style={{ float: "right", color: "green", fontWeight: "bold" }}>{props.product.price}€</span>
                                            </DialogContentText>
                                        </DialogContent>
                                        <DialogActions>
                                            <IconButton aria-label="close" color="primary" onClick={handleClose}>
                                                <CloseIcon />
                                            </IconButton>
                                        </DialogActions>
                                    </Dialog>
                                    {sessionStorage.getItem("tableId") ? (
                                        <div>
                                            {props.productType === 'Food' ? (
                                                <IconButton aria-label="add to favorites" color="primary" onClick={addFoodLikes}>
                                                <FavoriteIcon style={{ color: foodLike ? 'inherit' : 'gray' }} />
                                            </IconButton>
                                            ) : (
                                                <IconButton aria-label="add to favorites" color="primary" onClick={addDrinkLikes}>
                                                <FavoriteIcon style={{ color: drinkLike ? 'inherit' : 'gray' }} />
                                            </IconButton>
                                            )}
                                            <IconButton aria-label="add to order" color="primary" onClick={addToOrder}>
                                                <AddIcon />
                                            </IconButton>
                                        </div>
                                    ) : (
                                            null
                                        )}
                                </div>
                            </Grid>
                        </CardActions>
                    </Card>
                </div>
            ) : null}
        </div>
    )
}
export default Product