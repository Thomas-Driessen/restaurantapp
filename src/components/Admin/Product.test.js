import React from 'react';
import { render } from '@testing-library/react';
import Product from './Product';
import renderer from 'react-test-renderer';

it('Product renders correctly', () => {
  const tree = renderer
    .create(<Product />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Product', () => {
  var product = {
    id: 1,
    title: "Coca-Cola",
    price: 2.5,
    description: ""
  };
  const { getByText } = render(<Product product={product}/>);
  const linkElement = getByText(/Coca-cola/i);
  expect(linkElement).toBeInTheDocument();
  });