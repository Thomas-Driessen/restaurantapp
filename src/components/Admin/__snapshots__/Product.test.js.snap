// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <img
        alt="Image for Coca-Cola Not Found"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        height="250"
        style={
          Object {
            "height": 400,
          }
        }
        title="Coca-Cola"
      />
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-gutterBottom"
        >
          Coca-Cola
           
          <span
            style={
              Object {
                "color": "green",
                "float": "right",
              }
            }
          >
            2.5
            â‚¬
          </span>
        </h2>
        <h6
          className="MuiTypography-root MuiTypography-body1"
        />
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
        >
          <div
            style={
              Object {
                "alignItems": "right",
                "display": "flex",
              }
            }
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              target="_blank"
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Edit
                </span>
              </span>
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              target="_blank"
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Delete
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
